File: V_1/TravkingApplicationAPI/DTO/AddFeedback.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.OpenApi.Any;
using TravkingApplicationAPI.Models;

namespace TravkingApplicationAPI.DTO
{
    public class AddFeedback
    {
 public int feedbackId { get; set; }
    public int? totalAverageRating { get; set; }
    public object comments { get; set; }
    public string? description { get; set; }
    public UserModel? user { get; set; }
    public TaskModel? userTask { get; set; }

     public Module? Module { get; set; }
    public Rating?ratings{get;set;}
    public string? Submission_Count { get; set; }=null;




}

public class Module{
    public int ModuleId{get;set;}
    public string ModuleName{get;set;}
}

public class UserModel
{
    public int userId { get; set; }
    public string name { get; set; }
    public int total_Average_RatingStatus { get; set; }
}

public class TaskModel
{
    public int userTaskID { get; set; }
    public string taskName { get; set; }
}
}

File: V_1/TrackigApplicationFrontend/src/app/Models/add-feedback.ts
import { GetModules } from "./get-modules";

export class AddFeedback {
  feedbackId!: number;
  totalAverageRating?: number | null;
  comments?: any;
  description?: string | null;
  user?: UserModel | null;
  userTask?: TaskModel | null;
  module?: GetModules | null;
  ratings?: any[]|null;
  Submission_Count?: string | null;
}

interface UserModel {
  userId: number;
  name: string;
  total_Average_RatingStatus: number;
}

interface TaskModel {
  userTaskID: number;
  taskName: string;
}

interface Module {
  ModuleId: number;
  ModuleName: string;
}
